### DISCUSSION TITLE
osu! folder's location on macOS felt pretty unintuitive for me

### DISCUSSION THREAD

**Discussion by fyyrraaa** [ppy/osu#19093] [Ideas] [answered]
This might be very trivial, and probably a me-thing, but every time I want/need to dig up logs to report bugs (or add a custom ruleset) it always trips me up that it's under `~/.local/share`. Because I've used macOS for quite a long time, I would've expected it to be stored under `~/Library/Application Support`, which is its equivalent to Windows' `%APPDATA%` folder and Linux's various dot-prefixed folders under `$HOME` like `.local`, `.config`, etc. Some of the games that I could think of that does this are:
- Minecraft (https://minecraft.fandom.com/wiki/.minecraft#Locating_.minecraft)
- Aspyr's port of Windows games to Mac (https://support.aspyr.com/hc/en-us/articles/360056571831-How-to-find-my-Save-Files-and-Game-Folder)
- Jackbox Games (https://jackboxgames.happyfox.com/kb/article/29-how-to-find-and-clear-save-data-on-mac/)
- Games ported by Feral (one example is Tropico: https://support.feralinteractive.com/en/mac-linux-games/tropico4/faqs/saves/)

Outside of games, applications on macOS in general usually store its configuration files and user-specific files that aren't meant to be accessed directly by users under `~/Library` (for system-wide configurations it would be `/Library`).

I'm aware that there are outliers sadly (for instance Stardew Valley saves in `~/.config/StardewValley`) and it's more of a "de-facto" standard; the only definite (conceptual) documentation that I was able to dig up was an archived [documentation](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html), specifically under the section ["The Library Directory Stores App-Specific Files"](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html#//apple_ref/doc/uid/TP40010672-CH2-SW1). Moreover, apps that are published in the Mac App Store are stored in its own sandboxed file system akin to Flatpak (`~/Library/Containers/...`)

Would it be better for only new users to have their folder created under `~/Library/Application Support`? Or would it be better to also transparently move existing player's folder to that location? Knowing that in lazer the osu! folder isn't meant to be manipulated directly by us via file managers, instead being abstracted via the game.

**Comment by peppy** [comment]
We'll likely move it at some point, maybe. But in the mean time, you can open the folder by using the button at the top of the settings panel.

**Reply by fyyrraaa** [reply]
definitely! the thing that sparked this was when i came across ppy/osu-framework#18983, which meant that it's harder to access the logs for me because i forgot where it was stored under.

in the meantime i'll move it in to `~/Library/Application Support`. should this be marked as answered or should this be kept open?

**Comment by frenzibyte** [comment, marked as answer]
Will mention that Stardew Valley saving in `~/.config` is very likely to be related to https://github.com/dotnet/runtime/issues/40353, since that folder path on .NET is equivalent to `<user>/AppData/roaming` in Windows.

Gonna leave this unanswered, but not sure if there's any point for it if there's no issue thread tracking it.

**Reply by fyyrraaa** [reply]
should I create an issue from this discussion? or should I wait for the team to create it for me?

fyi there's an open PR at dotnet/runtime#68610 relating to this, just to doc things

**Reply by peppy** [reply]
We don't need to rely on that retrieval function, and it's not worth creating an issue for. We already have pretty good fallback logic which will allow us to prefer `~/.local/share` if already existing but default to `~/Library/Application Support` for new installs. I'd say PRing the new default path is a good start for futher testing.

**Reply by frenzibyte** [reply]
Have PR'd at https://github.com/ppy/osu-framework/pull/5304.
